src/main/java/com/hazelcast/aws/impl/AwsOperation.java

```diff
@@ -0,0 +1,332 @@
+package com.hazelcast.aws.impl;
+
+import com.hazelcast.aws.AwsConfig;
+import com.hazelcast.aws.exception.AwsConnectionException;
+import com.hazelcast.aws.security.Aws4RequestSigner;
+import com.hazelcast.aws.security.Aws4RequestSignerImpl;
+import com.hazelcast.aws.utility.Aws4RequestSignerUtils;
+import com.hazelcast.aws.security.AwsCredentials;
+import com.hazelcast.aws.utility.Environment;
+import com.hazelcast.aws.utility.MetadataUtil;
+import com.hazelcast.aws.utility.RetryUtils;
+import com.hazelcast.aws.utility.StringUtil;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.internal.json.Json;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.TimeZone;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.aws.utility.MetadataUtil.IAM_SECURITY_CREDENTIALS_URI;
+import static com.hazelcast.aws.utility.MetadataUtil.INSTANCE_METADATA_URI;
+import static com.hazelcast.aws.utility.StringUtil.isEmpty;
+import static com.hazelcast.aws.utility.StringUtil.isNotEmpty;
+import static com.hazelcast.nio.IOUtil.closeResource;
+
+/**
+ *
```

Comment:
any javadoc? does checkstyle pass without javadoc in class level?

Reply:
All Javadocs are TODO 👍